DevWs for GatorSystem LLC containing containing GatorSystem Motion Sensor

GatorSystem opened the pull request.

Commit:
DevWs for GatorSystem LLC containing containing GatorSystem Motion Sensor

KKlimczukS commented on devicetypes/gs/gatorsystem-homewatcher.src/gatorsystem-homewatcher.groovy:
```diff
@@ -0,0 +1,191 @@
+import physicalgraph.zigbee.clusters.iaszone.ZoneStatus
+
```
Unnecessary line break.

GatorSystem replied:
Deleted.

KKlimczukS commented on devicetypes/gs/gatorsystem-homewatcher.src/gatorsystem-homewatcher.groovy:
```diff
@@ -0,0 +1,191 @@
+	capability "Motion Sensor"
+	capability "Battery"
+	capability "Contact Sensor"
+    capability "Presence Sensor"
```
Please correct spacing (4 spaces -> 1 tab) - in a whole file.

GatorSystem replied:
Aligned.

KKlimczukS commented on devicetypes/gs/gatorsystem-homewatcher.src/gatorsystem-homewatcher.groovy:
```diff
@@ -0,0 +1,191 @@
+	capability "Contact Sensor"
+    capability "Presence Sensor"
+      
+    command "stopMotion"  
```
Custom commands are not supported in the new application. Please remove those 2 unnecessary lines.

GatorSystem replied:
Deleted.

KKlimczukS commented on devicetypes/gs/gatorsystem-homewatcher.src/gatorsystem-homewatcher.groovy:
```diff
@@ -0,0 +1,191 @@
+    command "stopMotion"  
+    command "resetOccupancy"
+      
+        fingerprint profileId: "0104", deviceId: "0402", inClusters: "0000, 0500", outClusters: "0502", manufacturer: "GatorSystem", model: "GSHW01"
```
1. deviceJoinName is missing in this fingerprint
Please add deviceJoinName: "GatorSystem Multipurpose Sensor" to the fingerprint
2. Please remove profileId, deviceId, inClusters, outClusters from the fingerprint
3. Please add "Raw Description" above this finger print in a comment
(It can be found at graph.api.smartthings.com ->  My Devices -> Raw Description) so it would look like:
// Raw Description .... 
fingerprint ....

GatorSystem replied:
Changed.

KKlimczukS commented on devicetypes/gs/gatorsystem-homewatcher.src/gatorsystem-homewatcher.groovy:
```diff
@@ -0,0 +1,191 @@
+        fingerprint profileId: "0104", deviceId: "0402", inClusters: "0000, 0500", outClusters: "0502", manufacturer: "GatorSystem", model: "GSHW01"
+      }
+
+       tiles(scale: 2) {
```
SmartThings application doesn't support tiles anymore. Please remove "tiles" section.

GatorSystem replied:
Deleted.

KKlimczukS commented on devicetypes/gs/gatorsystem-homewatcher.src/gatorsystem-homewatcher.groovy:
```diff
@@ -0,0 +1,191 @@
+    }
+}
+
+private List parseCatchAllMessage(String description) {
```
This method doesn't do anything.

KKlimczukS commented on devicetypes/gs/gatorsystem-homewatcher.src/gatorsystem-homewatcher.groovy:
```diff
@@ -0,0 +1,191 @@
+	Map map = [:]
+    List list = []
+	if (description?.startsWith('catchall:')) { //raw commands that smartthings does not or cannot interpret
+	  list = parseCatchAllMessage(description)    
```
I suggest to use zigbee.parseDescriptionAsMap(description)

GatorSystem replied:
Changed.

KKlimczukS commented on devicetypes/gs/gatorsystem-homewatcher.src/gatorsystem-homewatcher.groovy:
```diff
@@ -0,0 +1,191 @@
+	  list = parseCatchAllMessage(description)    
+	}
+	else if (description?.startsWith('read attr -')) {
+	  map = parseReportAttributeMessage(description)
```
The same as above, I suggest to use zigbee.parseDescriptionAsMap(description). 
It will handle "read attr - "

KKlimczukS commented on devicetypes/gs/gatorsystem-homewatcher.src/gatorsystem-homewatcher.groovy:
```diff
@@ -0,0 +1,191 @@
+	ZoneStatus zs = zigbee.parseZoneStatus(description)
+	Map resultMap = [:]
+
+    def seconds01 = 60
```
Wouldn't it look better if You renamed those variables to oneMinute and twoMinutes ?

KKlimczukS commented on devicetypes/gs/gatorsystem-homewatcher.src/gatorsystem-homewatcher.groovy:
```diff
@@ -0,0 +1,191 @@
+	}
+
+	log.debug "Parse returned map $map"
+    log.debug "Parse returned list $list"
```
This list might not be needed here.

KKlimczukS commented on devicetypes/gs/gatorsystem-homewatcher.src/gatorsystem-homewatcher.groovy:
```diff
@@ -0,0 +1,191 @@
+	return result
+}
+
+private Map parseReportAttributeMessage(String description) {
```
This method might not be needed if You used zigbee.parseDescriptionAsMap(description) in this DTH.

KKlimczukS commented on devicetypes/gs/gatorsystem-homewatcher.src/gatorsystem-homewatcher.groovy:
```diff
@@ -0,0 +1,191 @@
+    	resultMap = getMotionResult('active')  
+    } else if (zs.isBatterySet()){
+    	resultMap = getBatteryResult(10)
+    } else if (description.contains('0x8000')) {
```
Please use constants or a map to avoid magic numbers in the code (applies to all conditions in this method).

KKlimczukS commented on devicetypes/gs/gatorsystem-homewatcher.src/gatorsystem-homewatcher.groovy:
```diff
@@ -0,0 +1,191 @@
+
+private Map getBatteryResult(rawValue) {
+	log.debug "Battery rawValue = ${rawValue}"
+    sendEvent(name:"battery", value:rawValue)
```
createEvent should be used here (as a last line) and returned by this method.

KKlimczukS commented on devicetypes/gs/gatorsystem-homewatcher.src/gatorsystem-homewatcher.groovy:
```diff
@@ -0,0 +1,191 @@
+	if (value == 'active') {
+        log.debug 'intrusion'
+        String descriptionText = "{{ device.displayName }} detected intrusion"
+        return [
```
Please return createEvent( // Your map here) in every case.

KKlimczukS commented on devicetypes/gs/gatorsystem-homewatcher.src/gatorsystem-homewatcher.groovy:
```diff
@@ -0,0 +1,191 @@
+def installed() {
+	initialize()
+}
+def updated() {
```
This method will be run when user changes for example device name in the application.
Are You sure You want to call initialize() ? That method will send initial events (and those might be cause of triggering automations unnecessarily - if they will be defined by the user for this device).

Commit:
GS HomeWatcher DTH

GatorSystem committed:
Update gatorsystem-homewatcher.groovy

KKlimczukS commented on devicetypes/gs/gatorsystem-homewatcher.src/gatorsystem-homewatcher.groovy:
```diff
@@ -0,0 +1,109 @@
+        fingerprint deviceJoinName: "GatorSystem Multipurpose Sensor", manufacturer: "GatorSystem", model: "GSHW01"
+	}
+}
+def parse(String description) {
```
Please keep one line break between methods and sections (my previous remarks regarded only to separations having >= 2 line breaks)

KKlimczukS commented on devicetypes/gs/gatorsystem-homewatcher.src/gatorsystem-homewatcher.groovy:
```diff
@@ -0,0 +1,109 @@
+	capability "Battery"
+	capability "Contact Sensor"
+	capability "Presence Sensor"   
+	// Raw Description
```
Please add Raw Description content. As i mentioned before: 

It can be found at graph.api.smartthings.com -> My Devices -> Raw Description, so it would look like:
// Raw Description **insert content here**

Please let me know If You have any troubles finding it.

KKlimczukS commented on devicetypes/gs/gatorsystem-homewatcher.src/gatorsystem-homewatcher.groovy:
```diff
@@ -0,0 +1,109 @@
+def initialize() {
+	sendEvent(name:"motion", value:"inactive")
+	sendEvent(name:"battery", value:"100")
+        sendEvent(name:"presence", value:"not present")
```
Please fix all indentations (4 spaces -> 1 tab) applies to the whole file.

KKlimczukS commented on devicetypes/gs/gatorsystem-homewatcher.src/gatorsystem-homewatcher.groovy:
```diff
@@ -0,0 +1,109 @@
+	Map resultMap = [:]
+	def seconds01 = 60
+    	def seconds02 = 120
+	def resultOne = '0x8000'
```
Could You rename all those resultOne/Two/Three/Four/Five variables so they would be a bit more descriptive ?

GatorSystem committed:
Update gatorsystem-homewatcher.groovy

GatorSystem committed:
Update gatorsystem-homewatcher.groovy

I used GEDIT to align the code first and then copied it back to the edit window at Github. It looked aligned both in Gedit and in the Github window. So I'm trying to commit the aligned code again here.

GatorSystem committed:
Merge pull request #1 from GatorSystem/GatorSystem-patch-1-fixingAlignment

Update gatorsystem-homewatcher.groovy

GatorSystem commented:
I tried both Gedit and the edit window at Github. The code was aligned when I edited it. But after I committed and checked it again, the code became misaligned. Please offer me some tips on aligning the code. Thanks.

GatorSystem committed:
Update gatorsystem-homewatcher.groovy

I guess I figured it out; the misalignment was caused by tabs. I replaced all tabs with spaces.

GatorSystem commented:
Hi, I guess I figured it out. The misalignment was caused by tabs. I replaced all tabs with spaces. Now I understand your instruction "Please fix all indentations (4 spaces -> 1 tab) applies to the whole file.".

GatorSystem commented:
Hello, Konrad,

I think I successfully addressed the issues.

Best regards,
Ying

On Mon, Nov 9, 2020 at 4:15 AM Konrad K <notifications@github.com> wrote:

> *@KKlimczukS* requested changes on this pull request.
> ------------------------------
>
> In
> devicetypes/gs/gatorsystem-homewatcher.src/gatorsystem-homewatcher.groovy
> <https://github.com/SmartThingsCommunity/SmartThingsPublic/pull/46565#discussion_r519681857>
> :
>
> > @@ -0,0 +1,109 @@
> +import physicalgraph.zigbee.clusters.iaszone.ZoneStatus
> +metadata {
> +	definition (name: "GatorSystem HomeWatcher", namespace: "GS", author: "GS_coder000") {
> +	capability "Motion Sensor"
> +	capability "Battery"
> +	capability "Contact Sensor"
> +	capability "Presence Sensor"
> +	// Raw Description
> +        fingerprint deviceJoinName: "GatorSystem Multipurpose Sensor", manufacturer: "GatorSystem", model: "GSHW01"
> +	}
> +}
> +def parse(String description) {
>
> Please keep one line break between methods and sections (my previous
> remarks regarded only to separations having >= 2 line breaks)
> ------------------------------
>
> In
> devicetypes/gs/gatorsystem-homewatcher.src/gatorsystem-homewatcher.groovy
> <https://github.com/SmartThingsCommunity/SmartThingsPublic/pull/46565#discussion_r519685245>
> :
>
> > @@ -0,0 +1,109 @@
> +import physicalgraph.zigbee.clusters.iaszone.ZoneStatus
> +metadata {
> +	definition (name: "GatorSystem HomeWatcher", namespace: "GS", author: "GS_coder000") {
> +	capability "Motion Sensor"
> +	capability "Battery"
> +	capability "Contact Sensor"
> +	capability "Presence Sensor"
> +	// Raw Description
>
> Please add Raw Description content. As i mentioned before:
>
> It can be found at graph.api.smartthings.com -> My Devices -> Raw
> Description, so it would look like:
> // Raw Description *insert content here*
>
> Please let me know If You have any troubles finding it.
> ------------------------------
>
> In
> devicetypes/gs/gatorsystem-homewatcher.src/gatorsystem-homewatcher.groovy
> <https://github.com/SmartThingsCommunity/SmartThingsPublic/pull/46565#discussion_r519686595>
> :
>
> > +        String descriptionText = "{{ device.displayName }} detected window closed"
> +        createEvent(
> +            name: 'contact',
> +            value: "closed",
> +            descriptionText: descriptionText,
> +            translatable: true
> +	)
> +    }
> +}
> +def installed() {
> +	initialize()
> +}
> +def initialize() {
> +	sendEvent(name:"motion", value:"inactive")
> +	sendEvent(name:"battery", value:"100")
> +        sendEvent(name:"presence", value:"not present")
>
> Please fix all indentations (4 spaces -> 1 tab) applies to the whole file.
> ------------------------------
>
> In
> devicetypes/gs/gatorsystem-homewatcher.src/gatorsystem-homewatcher.groovy
> <https://github.com/SmartThingsCommunity/SmartThingsPublic/pull/46565#discussion_r519694114>
> :
>
> > +	}
> +	else if (description?.startsWith('zone status')) {
> +	  map = parseIasMessage(description)
> +	}
> +	log.debug "Parse returned map $map"
> +	if (map != null) {
> +		def result = map ? createEvent(map) : null
> +        	return result
> +	}
> +}
> +private Map parseIasMessage(String description) {
> +	ZoneStatus zs = zigbee.parseZoneStatus(description)
> +	Map resultMap = [:]
> +	def seconds01 = 60
> +    	def seconds02 = 120
> +	def resultOne = '0x8000'
>
> Could You rename all those resultOne/Two/Three/Four/Five variables so they
> would be a bit more descriptive ?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/SmartThingsCommunity/SmartThingsPublic/pull/46565#pullrequestreview-526088197>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQUWVRQSCFRZBI7BJULODV3SO66KXANCNFSM4SE5DWIQ>
> .
>

GatorSystem committed:
Update gatorsystem-homewatcher.groovy

I tested the code with a device and made changes to make the code work better.

KKlimczukS commented on devicetypes/gs/gatorsystem-homewatcher.src/gatorsystem-homewatcher.groovy:
```diff
@@ -0,0 +1,130 @@
+    capability "Battery"
+    capability "Contact Sensor"
+    capability "Presence Sensor"   
+    // Raw Description 08 0104 0402 00 02 0000 0500 01 0502
```
Please add a line break above this line.

KKlimczukS commented on devicetypes/gs/gatorsystem-homewatcher.src/gatorsystem-homewatcher.groovy:
```diff
@@ -0,0 +1,130 @@
+    if (description?.startsWith('catchall:')) { //raw commands that smartthings does not or cannot interpret
+        map = zigbee.parseDescriptionAsMap(description)   
+    }
+    else if (description?.startsWith('read attr -')) {
```
please move this "else if" statement to the upper line so it looks: "} else if (...) {"

KKlimczukS commented on devicetypes/gs/gatorsystem-homewatcher.src/gatorsystem-homewatcher.groovy:
```diff
@@ -0,0 +1,130 @@
+    else if (description?.startsWith('read attr -')) {
+        map = zigbee.parseDescriptionAsMap(description)
+    }
+    else if (description?.startsWith('zone status')) {
```
please move this "else if" statement to the upper line so it looks: "} else if (...) {"

KKlimczukS reviewed (changes requested):
> Hi, I guess I figured it out. The misalignment was caused by tabs. I replaced all tabs with spaces. Now I understand your instruction "Please fix all indentations (4 spaces -> 1 tab) applies to the whole file.".

What I really meant was to change change 4 spaces to 1 tab.

GatorSystem committed:
Update gatorsystem-homewatcher.groovy

Revised per Konrad's comments.

GatorSystem committed:
Update gatorsystem-homewatcher.groovy

Made small changes to logged information.

KKlimczukS commented on devicetypes/gs/gatorsystem-homewatcher.src/gatorsystem-homewatcher.groovy:
```diff
@@ -0,0 +1,131 @@
+    if (zs.isAlarm1Set()) {
+        runIn(seconds02, stopMotion) 
+        resultMap = getMotionResult('active')  
+    } else if (zs.isBatterySet()){
```
Please add space between parentheses and brace (as it's in other conditions).

KKlimczukS commented on devicetypes/gs/gatorsystem-homewatcher.src/gatorsystem-homewatcher.groovy:
```diff
@@ -0,0 +1,131 @@
+private Map parseIasMessage(String description) {
+    ZoneStatus zs = zigbee.parseZoneStatus(description)
+    Map resultMap = [:]
+    def seconds01 = 60
```
Wouldn't it look better if those variables were renamed to oneMinute and twoMinutes ?

GatorSystem committed:
Update gatorsystem-homewatcher.groovy

Deleted a line break.

KKlimczukS commented on devicetypes/gs/gatorsystem-homewatcher.src/gatorsystem-homewatcher.groovy:
```diff
@@ -0,0 +1,131 @@
+import physicalgraph.zigbee.clusters.iaszone.ZoneStatus
+
+metadata {
+    definition (name: "GatorSystem HomeWatcher", namespace: "GS", author: "GS_coder000") {
```
I think we still have the issues with spacing in this file (changing 4 spaces to 1 tab would fix it).

KKlimczukS commented on devicetypes/gs/gatorsystem-homewatcher.src/gatorsystem-homewatcher.groovy:
```diff
@@ -0,0 +1,130 @@
+        map = parseIasMessage(description)
+    }
+    log.debug "Parse returned map $map"
+    if (map != null) {    
```
Wouldn't it look better if You returned createEvent(map) here and added else {return null} statement ?

KKlimczukS replied:
I rethinked that,  please get rid of that last 'else {return null}'  statement, it's redundant.

KKlimczukS commented on devicetypes/gs/gatorsystem-homewatcher.src/gatorsystem-homewatcher.groovy:
```diff
@@ -0,0 +1,130 @@
+}
+
+def stopMotion() {
+    def seconds = 120
```
If You use this variable in a log message only - wouldn't it be better to write that value explicitly in that message ?

KKlimczukS commented on devicetypes/gs/gatorsystem-homewatcher.src/gatorsystem-homewatcher.groovy:
```diff
@@ -0,0 +1,130 @@
+}
+
+def resetOccupancy() {
+    def seconds = 60
```
If You use this variable in a log message only - wouldn't it be better to write that value explicitly in that message ?

KKlimczukS commented on devicetypes/gs/gatorsystem-homewatcher.src/gatorsystem-homewatcher.groovy:
```diff
@@ -0,0 +1,130 @@
+    capability "Presence Sensor"   
+        
+    // Raw Description 08 0104 0402 00 02 0000 0500 01 0502
+    fingerprint deviceJoinName: "GatorSystem Multipurpose Sensor", manufacturer: "GatorSystem", model: "GSHW01"
```
Please place deviceJoinName after manufacturer.

KKlimczukS replied:
I should have asked You in a first place to move deviceJoinName **after a manufacturer and a model.** 
Please, could You do that ?

KKlimczukS reviewed (commented):
> I tried both Gedit and the edit window at Github. The code was aligned when I edited it. But after I committed and checked it again, the code became misaligned. Please offer me some tips on aligning the code. Thanks.

Did You try to fix the code in other code editor/ IDE and then paste it in the github ?

GatorSystem committed:
Update gatorsystem-homewatcher.groovy

Revised per Konrad's comments.

GatorSystem committed:
Update gatorsystem-homewatcher.groovy

Minor revisions to log information.

KKlimczukS reviewed (changes requested):
Two small changes and I think it's gonna be ready to be merged.

GatorSystem committed:
Update gatorsystem-homewatcher.groovy

Revised per Conrad's comments.

GatorSystem commented:
Thanks, Konrad.
